libraries/libfc/include/fc/crypto/xxh3.hpp

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#include <fc/fwd.hpp>
+#include <fc/crypto/packhash.hpp>
+#include <fc/io/raw_fwd.hpp>
+
+#include <xxHash/xxhash.h>
+
+namespace fc {
+
+namespace detail {
+struct alignas(64) alignmeto64 {};
+
+template <typename T>
+concept ContiguousCharSource = requires(const T& obj) {
+   { obj.data() } -> std::convertible_to<const char*>;
+   { obj.size() } -> std::unsigned_integral;
+};
+}
```

Comment:
We really need a better place for stuff like this -- even just putting in `detail` here doesn't protect against against ODR oopsies since it's just in `fc::detail` still.

Reply:
Put in a separate file with a different name space?